platform/cxxsupport/mstd_span

```diff
@@ -0,0 +1,365 @@
+    {
+        MBED_ASSERT(offset <= size() && (count == dynamic_extent || count <= size() - offset ));
+        return span<element_type, dynamic_extent>(data() + offset, count == dynamic_extent ? size() - offset : count );
+    }
+
+    // Observers
+    constexpr index_type size() const noexcept
+    {
+        return _storage._size;
+    }
+
+    constexpr index_type size_bytes() const noexcept
+    {
+        return size() * sizeof(element_type);
+    }
+
+    constexpr bool empty() const noexcept
+    {
+        return size() == 0;
+    }
+
+    // Element access
+    constexpr reference operator[](index_type idx) const
+    {
+        MBED_ASSERT(idx < size());
+        return *(data() + idx);
+    }
+
+    constexpr reference front() const
+    {
+        MBED_ASSERT(not empty());
+        return *data();
+    }
+
+    constexpr reference back() const
+    {
+        MBED_ASSERT(not empty());
+        return *(data() + (size() - 1));
+    }
+
+    constexpr pointer data() const noexcept
+    {
+        return _storage._data;
+    }
+
+    // Iterators
+    constexpr iterator begin() const noexcept
+    {
+        return data();
+    }
+
+    constexpr iterator end() const noexcept
+    {
+        return data() + size();
+    }
+
+    constexpr reverse_iterator rbegin() const noexcept
+    {
+        return reverse_iterator(end());
+    }
+
+    constexpr reverse_iterator rend() const noexcept
+    {
+        return reverse_iterator(begin());
+    }
+
+private:
+    detail::storage<element_type, extent> _storage;
+};
+
+template<class ElementType, size_t Extent>
+span<const unsigned char, Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent>
+as_bytes(span<ElementType, Extent> s) noexcept
+{
+    return {reinterpret_cast<const unsigned char*>(s.data()), s.size_bytes()};
+}
+
+template<class ElementType, size_t Extent>
+span<unsigned char, Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent>
+as_writable_bytes(span<ElementType, Extent> s) noexcept
+{
+    static_assert(not is_const<ElementType>::value);
+    return {reinterpret_cast<unsigned char*>(s.data()), s.size_bytes()};
+}
+
+template<class ElementType, size_t Extent>
+constexpr span<ElementType, Extent> make_span(ElementType (&arr)[Extent])
+{
+    return span<ElementType, Extent>(arr);
+}
+
+template<class ElementType, size_t Extent>
+constexpr span<const ElementType, Extent> make_span(const ElementType (&arr)[Extent])
+{
+    return span<const ElementType, Extent>(arr);
+}
+
+template<class ElementType, size_t Extent>
+constexpr span<ElementType, Extent> make_span(array<ElementType, Extent>& arr)
+{
+    return span<ElementType, Extent>(arr);
+}
+
+template<class ElementType, size_t Extent>
+constexpr span<const ElementType, Extent> make_span(const array<ElementType, Extent>& arr)
+{
+    return span<const ElementType, Extent>(arr);
+}
+
+template<class R>
+constexpr span<typename R::value_type> make_span(R& cont)
+{
+    return {cont};
```

Comment:
Are the `{}` doing anything here? Isn't it invoking an implicit converting constructor with or without?